indico/modules/events/registration/client/js/form/fields/DateInput.jsx

```diff
@@ -223,29 +224,32 @@ export function DateSettings() {
     {key: '24h', value: '24h', text: Translate.string('24 hours')},
   ];
   return (
-    <Form.Group widths="equal">
-      <FinalDropdown
-        name="dateFormat"
-        label={Translate.string('Date format')}
-        options={dateOptions}
-        required
-        selection
-        fluid
-      />
-      <Field name="dateFormat" subscription={{value: true}}>
-        {({input: {value: dateFormat}}) => (
-          <FinalDropdown
-            name="timeFormat"
-            label={Translate.string('Time format')}
-            options={timeOptions}
-            placeholder={Translate.string('None')}
-            disabled={!dateFormat.includes('%d')}
-            parse={p.nullIfEmpty}
-            selection
-            fluid
-          />
-        )}
-      </Field>
-    </Form.Group>
+    <div>
+      <Form.Group widths="equal">
+        <FinalDropdown
+          name="dateFormat"
+          label={Translate.string('Date format')}
+          options={dateOptions}
+          required
+          selection
+          fluid
+        />
+        <Field name="dateFormat" subscription={{value: true}}>
+          {({input: {value: dateFormat}}) => (
+            <FinalDropdown
+              name="timeFormat"
+              label={Translate.string('Time format')}
+              options={timeOptions}
+              placeholder={Translate.string('None')}
+              disabled={!dateFormat.includes('%d')}
+              parse={p.nullIfEmpty}
+              selection
+              fluid
+            />
+          )}
+        </Field>
+      </Form.Group>
+      {renderPluginComponents('date-input-settings', {...props})}
```

Comment:
this name should be specific to the regform. e.g. `regform-date-input-settings`

also, what kind of settings are you adding? maybe they could be useful upstream (if it's stuff like minimum age, minimum year, etc)

Reply:
I just use "htmlName" from props in the plugin to check which field is, as we add some settings only for birth date field.
I can of course pass only htmlName to the plugin, but I prefer to keep it wide if later on we need to do some more actions on that field...

Reply:
Plugin:
if(props.htmlName === 'birth_date')
  return(...)
else
  return(...)

Reply:
<img width="488" alt="date-settings-un" src="https://github.com/indico/indico/assets/11231950/1d13297d-38cd-4c5c-8ed3-e992c7f9145e">

Reply:
Now I'm curious: you say "recommended" age - does that mean it's not enforced, ie you can register even with a lower age?

Anyway, the way I could imagine this in the core (I have not yet discussed this w/ the others, so just my own thought): When selecting a date format that contains only date+month+year (ie no time, since that makes no sense), an option to require a minimum age (like in your screenshot) would be displayed (fine to allow overriding a default message there as well, the default could be something as simple as "you must be at least x years old"). Also, I would not use the date of the registration but rather the start date of the event as the date to determine the age...

Reply:
How it works in UN:
Security/Managers set a default minimum age in the settings,
when user registers, if the age calculation (current date - date input without time) is lower than recommended age, a warning message is shown in the registration form,
it's just warning, it doesn't not prevent to submit the registration.
The warning message you see is the one defined by ours users, if implemented in core, we can use your message as they can always modify it.

Reply:
OK maybe let's keep it in the plugin for now, since it probably needs quite a few settings otherwise